pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,309 @@
+use crate::pallet::*;
+use frame_support::{pallet_prelude::*, traits::{Currency, ReservableCurrency, LockableCurrency, WithdrawReasons}};
+use sp_runtime::{traits::{AccountIdConversion, Zero}, Perbill, Saturating};
+use subsocial_support::SpaceId;
+
+impl<T: Config> Pallet<T> {
+    /// Calculate the creator reward for the specified era.
+    /// If successfull, returns reward amount.
+    /// In case reward cannot be claimed or was already claimed, an error is raised.
+    pub(super) fn calculate_creator_reward(
+        creator_stake_info: &CreatorStakeInfo<BalanceOf<T>>,
+        creator_info: &CreatorInfo<T::AccountId>,
+        era: EraIndex,
+    ) -> Result<BalanceOf<T>, Error<T>> {
+        let current_era = Self::current_era();
+        if let CreatorState::Unregistered(unregister_era) = creator_info.state {
+            ensure!(era < unregister_era, Error::<T>::NotRegisteredCreator);
+        }
+        ensure!(era < current_era, Error::<T>::EraOutOfBounds);
+
+        ensure!(
+                !creator_stake_info.creator_reward_claimed,
+                Error::<T>::AlreadyClaimedInThisEra,
+            );
+        ensure!(
+                creator_stake_info.total > Zero::zero(),
+                Error::<T>::NotStakedCreator,
+            );
+
+        let rewards_and_stakes =
+            Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
+
+        // Calculate the creator reward for this era.
+        let (creator_reward, _) = Self::distributed_rewards_between_creator_and_stakers(creator_stake_info, &rewards_and_stakes);
+
+        Ok(creator_reward)
+    }
+
+    /// `Err` if pallet disabled for maintenance, `Ok` otherwise
+    pub fn ensure_pallet_enabled() -> Result<(), Error<T>> {
```

Comment:
Why are we not calling this func in `unregister_creator()` and `force_unregister_creator()`?